tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,1871 @@
+        self.execute_command(f'routerdowngradethreshold {threshold}')
+
+    def get_router_eligible(self) -> bool:
+        """Indicates whether the router role is enabled or disabled."""
+        return self.__parse_Enabled_or_Disabled(self.execute_command('routereligible'))
+
+    def enable_router_eligible(self):
+        """Disable the router role."""
+        self.execute_command('routereligible enable')
+
+    def disable_router_eligible(self):
+        """Disable the router role."""
+        self.execute_command('routereligible disable')
+
+    def get_router_list(self) -> List[RouterId]:
+        """Get allocated Router IDs."""
+        line = self.__parse_str(self.execute_command('router list'))
+        return list(map(RouterId, line.strip().split()))
+
+    def get_router_table(self) -> Dict[RouterId, Dict[str, Any]]:
+        """table of routers."""
+        output = self.execute_command('router table')
+        if len(output) < 2:
+            raise UnexpectedCommandOutput(output)
+
+        #
+        # Example output:
+        #
+        # | ID | RLOC16 | Next Hop | Path Cost | LQ In | LQ Out | Age | Extended MAC     |
+        # +----+--------+----------+-----------+-------+--------+-----+------------------+
+        # | 21 | 0x5400 |       21 |         0 |     3 |      3 |   5 | d28d7f875888fccb |
+        # | 56 | 0xe000 |       56 |         0 |     0 |      0 | 182 | f2d92a82c8d8fe43 |
+        # Done
+        #
+
+        table = {}
+        for line in output[2:]:
+            line = line.strip()
+            if not line:
+                continue
+
+            fields = line.split('|')
+            _, id, rloc16, next_hop, path_cost, lq_in, lq_out, age, extaddr, _ = fields
+
+            table[RouterId(id)] = {
+                'id': RouterId(id),
+                'rloc16': Rloc16(rloc16, 16),
+                'next_hop': int(next_hop),
+                'path_cost': int(path_cost),
+                'lq_in': int(lq_in),
+                'lq_out': int(lq_out),
+                'age': int(age),
+                'extaddr': extaddr,
+            }
+
+        return table
+
+    # TODO: router <id>
+
+    #
+    # Router utilities: Child management
+    #
+
+    def get_child_table(self) -> Dict[ChildId, Dict[str, Any]]:
+        """Get the table of attached children."""
+        output = self.execute_command('child table')
+        if len(output) < 2:
+            raise UnexpectedCommandOutput(output)
+
+        #
+        # Example output:
+        # | ID  | RLOC16 | Timeout    | Age        | LQ In | C_VN |R|D|N|Ver|CSL|QMsgCnt| Extended MAC     |
+        # +-----+--------+------------+------------+-------+------+-+-+-+---+---+-------+------------------+
+        # |   1 | 0xc801 |        240 |         24 |     3 |  131 |1|0|0|  3| 0 |     0 | 4ecede68435358ac |
+        # |   2 | 0xc802 |        240 |          2 |     3 |  131 |0|0|0|  3| 1 |     0 | a672a601d2ce37d8 |
+        # Done
```

Comment:
Can you please add support for previous version below? Sam for neighbour table.

| ID  | RLOC16 | Timeout    | Age        | LQ In | C_VN |R|D|N| Extended MAC     |

Reply:
it's becoming more and more difficult to handle all versions :(

Reply:
handled